adapters/Maio/MaioAdapter/GADMMaioError.m

```diff
@@ -11,10 +11,14 @@
 @implementation GADMMaioError
 
 + (NSError *)errorWithDescription:(NSString *)description {
+  return [self errorWithDescription:description errorCode:0];
+}
+
++ (NSError *)errorWithDescription:(NSString *)description errorCode:(NSInteger)errorCode {
   description = [description copy];
   NSDictionary *userInfo =
       @{NSLocalizedDescriptionKey : description, NSLocalizedFailureReasonErrorKey : description};
-  NSError *error = [NSError errorWithDomain:kGADMMaioErrorDomain code:0 userInfo:userInfo];
+  NSError *error = [NSError errorWithDomain:kGADMMaioErrorDomain code:errorCode userInfo:userInfo];
   return error;
```

Comment:
We are currently updating our adapter to use granular error codes to aid debugging. One of these changes include passing in the `error` object (if any) as is if the maio SDK provides one, or introducing an integer ENUM with error codes specific to a failure case.
Can you update the PR to include these changes as well?

You can refer to this commit for reference: https://github.com/googleads/googleads-mobile-ios-mediation/commit/d607473ebf75357a80a570a1118959e52862f687#diff-6c81779a9805e2b33d4cd73aba766c8f818784ca4e9075ef536b3d9be80fa4f8